include/BSMPT/gravitational_waves/gw.h

```diff
@@ -0,0 +1,196 @@
+// Copyright (C) 2024 Lisa Biermann, Margarete Mühlleitner, Rui Santos, João
+// Viana
+//
+// SPDX-FileCopyrightText: 2024 Lisa Biermann, Margarete Mühlleitner, Rui
+// Santos, João Viana
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+#ifndef INCLUDE_BSMPT_GW_H_
+#define INCLUDE_BSMPT_GW_H_
+
+/**
+ * @file gravitational wave calculation
+ */
+
+#include <BSMPT/bounce_solution/bounce_solution.h> // BounceSolution
+#include <BSMPT/utility/Logger.h>
+#include <math.h> // M_PI
+
+namespace BSMPT
+{
+
+/**
+ * @brief struct to store all calculated GW data
+ */
+struct GravitationalWaveData
+{
+  bool swON             = true;  // enable sound wave contribution by default
+  bool turbON           = true;  // enable turbulence contribution by default
+  double transitionTemp = false; // transition temperature
+  double PTStrength     = false; // strength of EW phase transition
+  double InvTimeScale   = false; // inverse time scale beta/H
+  double vb             = false; // bubble wall velocity
+
+  double kappa_sw = false;   // efficiency factor for sound waves
+  double K_sw     = false;   // kinetic energy fraction in fluid of total bubble
+                             // energy for sound waves
+  double HR         = false; // time scale times max. velocity for sound waves
+  double kappa_turb = false; // efficiency factor for turbulence
+  double K_turb     = false; // kinetic energy fraction in fluid of total bubble
+                             // energy for turbulence
+  double gstar = false;      // number of eff. d.o.f.
+  double Hstar = false;      // Hubble rate at percolation temperature
+
+  double fPeakSoundWave        = false; // peak frequency for sound wave
+  double h2OmegaPeakSoundWave  = false; // peak amplitude for sound wave
+  double fPeakTurbulence       = false; // peak frequency for turbulence
+  double h2OmegaPeakTurbulence = false; // peak amplitude for turbulence
+
+  std::string status = "not_set"; // gw calculation status
```

Comment:
Is this some placeholder or should this string only have specific values? Seems like the second one

Reply:
This string only has specific values and if not yet set, is set to "not_set".

Reply:
Then you could make an enum out of it or why should it be a string?

Reply:
Done.